src/main/java/org/red5/server/net/rtmp/event/Ping.java

```diff
@@ -296,7 +296,7 @@ public void writeExternal(ObjectOutput out) throws IOException {
 		switch (eventType) {
 			case PING_CLIENT:
 			case PONG_SERVER:
-				out.writeDouble(value2.intValue());
+				out.writeInt(value2.intValue());

```

Comment:
The ping and pong types expect an int per the spec; they are the only types handled by our Ping that require this for value 2.